src/Memeorr.ts

```diff
@@ -340,3 +328,387 @@ ponder.on("MemeCelo:Unleashed", async ({ event, context }) => {
+    },
+  });
+  try {
+    await context.db.Heart.create({
+      id: event.log.id,
+      data: {
+        chain: "celo",
+        hearter: event.args.hearter,
+        memeNonce: event.args.memeNonce,
+        amount: event.args.amount,
+        timestamp: Number(event.block.timestamp),
+        blockNumber: Number(event.block.number),
+      },
+    });
+
+    // cannot specifiy memeNonce as unique, opt for findMany and limit 1
+    const memeToken = await context.db.MemeToken.findMany({
+      where: {
+        nonce: event.args.memeNonce,
+        chain: "base",        
+      },
+      limit: 1,
+    }).then(res => res.items[0]); // pull first item
+
+    if(!memeToken) return;
+
+    await context.db.MemeToken.update({
+      id: memeToken.id,
+      data: {
+        heartCount: (memeToken?.heartCount || 0n) + 1n,
+      },
+    });
+
+    const hearterId = `${event.args.hearter}`;
+    const existingTotal = await context.db.totalHeartAmount.findUnique({
+      id: hearterId,
+    });
+
+    if (existingTotal) {
+      await context.db.totalHeartAmount.update({
+        id: hearterId,
+        data: {
+          amount: existingTotal.amount + event.args.amount,
+        },
+      });
+    } else {
+      await context.db.totalHeartAmount.create({
+        id: hearterId,
+        data: {
+          chain: "celo",
+          amount: event.args.amount,
+          nonce: event.args.memeNonce,
+        },
+      });
+    }
+  } catch (e) {
+    console.log(e);
+  }
+});
+
+ponder.on("MemeBase_0_2_0:OLASJourneyToAscendance", async ({ event, context }) => {
+  await context.db.OLASJourneyToAscendanceEvent.create({
+    id: event.log.id,
+    data: {
+      chain: "base",
+      amount: event.args.amount,
+      timestamp: Number(event.block.timestamp),
+      blockNumber: Number(event.block.number),
+    },
+  });
+});
+
+ponder.on("MemeCelo_0_2_0:OLASJourneyToAscendance", async ({ event, context }) => {
+  await context.db.OLASJourneyToAscendanceEvent.create({
+    id: event.log.id,
+    data: {
+      chain: "celo",
+      amount: event.args.amount,
+      timestamp: Number(event.block.timestamp),
+      blockNumber: Number(event.block.number),
+    },
+  });
+});
+
+ponder.on("MemeBase_0_2_0:Purged", async ({ event, context }) => {
+  await context.db.PurgeEvent.create({
+    id: event.log.id,
+    data: {
+      chain: "base",
+      memeToken: event.args.memeToken,
+      timestamp: Number(event.block.timestamp),
+      blockNumber: Number(event.block.number),
+    },
+  });
+});
+
+ponder.on("MemeCelo_0_2_0:Purged", async ({ event, context }) => {
+  await context.db.PurgeEvent.create({
+    id: event.log.id,
+    data: {
+      chain: "celo",
+      memeToken: event.args.memeToken,
+      timestamp: Number(event.block.timestamp),
+      blockNumber: Number(event.block.number),
+    },
+  });
+});
+
+ponder.on("MemeBase_0_2_0:Summoned", async ({ event, context }) => {
+  try {
+    await context.db.totalHeartAmount.create({
```

Comment:
review `id`

Reply:
summoners can summon multiple times, not sure this makes sense